openfl-tutorials/interactive_api/PyTorch_MedMNIST_3D/workspace/Pytorch_MedMNIST_3D.ipynb

```diff
@@ -0,0 +1,930 @@
+      "      (shortcut): Sequential()\n",
+      "    )\n",
+      "  )\n",
+      "  (avgpool): AdaptiveAvgPool3d(output_size=(1, 1, 1))\n",
+      "  (linear): Linear(in_features=512, out_features=2, bias=True)\n",
+      ")\n",
+      ")\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(model)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8d1c78ee",
+   "metadata": {},
+   "source": [
+    "### Register model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 19,
+   "id": "59831bcd",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from copy import deepcopy\n",
+    "\n",
+    "framework_adapter = 'openfl.plugins.frameworks_adapters.pytorch_adapter.FrameworkAdapterPlugin'\n",
+    "MI = ModelInterface(model=model.model, optimizer=optimizer, framework_plugin=framework_adapter)\n",
+    "\n",
+    "# Save the initial model state\n",
+    "initial_model = deepcopy(model.model)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "849c165b",
+   "metadata": {},
+   "source": [
+    "## Define and register FL tasks"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 22,
+   "id": "4ff463bd",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "TI = TaskInterface()\n",
+    "\n",
+    "from logging import getLogger\n",
+    "\n",
+    "import torch\n",
+    "import tqdm\n",
+    "\n",
+    "logger = getLogger(__name__)\n",
+    "\n",
+    "train_custom_params={'criterion':criterion,'task':task}\n",
+    "\n",
+    "# Task interface currently supports only standalone functions.\n",
+    "@TI.add_kwargs(**train_custom_params)\n",
+    "@TI.register_fl_task(model='model', data_loader='train_loader',\n",
+    "                     device='device', optimizer='optimizer')\n",
+    "def train(model, train_loader, device, optimizer, criterion, task):\n",
+    "    total_loss = []\n",
+    "    \n",
+    "    train_loader = tqdm.tqdm(train_loader, desc=\"train\")\n",
+    "    model.train()\n",
+    "    model.to(device)\n",
+    "    for inputs, targets in train_loader:\n",
+    "        \n",
+    "        inputs, targets = inputs.to(device), targets.to(device)\n",
+    "    \n",
+    "        optimizer.zero_grad()\n",
+    "        outputs = model(inputs.to(device))\n",
+    "        \n",
+    "        if task == 'multi-label, binary-class':\n",
+    "            targets = targets.to(torch.float32).to(device)\n",
+    "            loss = criterion(outputs, targets)\n",
+    "        else:\n",
+    "            targets = torch.squeeze(targets, 1).long().to(device)\n",
+    "            loss = criterion(outputs, targets)\n",
+    "        \n",
+    "        total_loss.append(loss.item())\n",
+    "        \n",
+    "        loss.backward()\n",
+    "        optimizer.step()\n",
+    "    \n",
+    "    return {'train_loss': np.mean(total_loss),}\n",
+    "\n",
+    "\n",
+    "\n",
+    "val_evaluator = medmnist.Evaluator(data_flag, 'val')\n",
```

Comment:
One can either import tqdm and use it with tqdm.tqdm(..) or by just importing tqdm from tqdm. Since the rest of the code is using the firts, I've just replaced the import with your recommended one